### PR TITLE
RM-8951 Add test to ensure the available resources are set correctly on re-motion assemblies

### PR DESCRIPTION
**Pull request by a-ctor**
https://re-motion.atlassian.net/browse/RM-8951

Draft PR to review the current implementation. Once the design is approved I will add the individual test for all the assemblies.

### CODE REVIEW

File: Remotion/ObjectBinding/UnitTests/AvailableResourcesTest.cs
Comment (MichaelKetting) [inline Remotion/ObjectBinding/UnitTests/AvailableResourcesTest.cs:29]: why not hardcode the attribute?

File: Remotion/ObjectBinding/UnitTests/ObjectBinding.UnitTests.csproj
Comment (MichaelKetting) [inline Remotion/ObjectBinding/UnitTests/ObjectBinding.UnitTests.csproj:15]: THere's a different include pattern in line 14. I think this needs chaing in line 15?
-> `SharedSource\NUnit\UnitTesting\Globalization` would be the location for the shared file, also in the sharedsource location.

File: SharedSource/Development/Globalization/AvailableResourcesTester.cs
Comment (MichaelKetting) [inline SharedSource/Development/Globalization/AvailableResourcesTester.cs:16]: not yet reviewed

**Review by MichaelKetting — CHANGES_REQUESTED** [review]

File: Remotion/ObjectBinding/UnitTests/AvailableResourcesTest.cs
Comment (a-ctor) [reply Remotion/ObjectBinding/UnitTests/AvailableResourcesTest.cs:29]: Because I did not want to introduce a globalization reference in the shared source project and passing the attribute seems like reasonable compromise that won't break instantly on rename/move of the attribute

Comment (MichaelKetting) [reply Remotion/ObjectBinding/UnitTests/AvailableResourcesTest.cs:29]: Ah, I see. hmm... got an idea; we could link the Attribute over into the shared files project. That way, we can simplify the code :)

File: Remotion/ObjectBinding/UnitTests/ObjectBinding.UnitTests.csproj
Comment (MichaelKetting) [inline Remotion/ObjectBinding/UnitTests/ObjectBinding.UnitTests.csproj:14]: check. ok.

File: SharedSource/Development/NUnit/UnitTesting/Globalization/AvailableResourcesTester.cs
Comment (MichaelKetting) [inline SharedSource/Development/NUnit/UnitTesting/Globalization/AvailableResourcesTester.cs:19]: waiting on the final decision if linking the attribute into the dev project would make sense or not.

**Review by MichaelKetting — CHANGES_REQUESTED** [review]

File: SharedSource/Development/NUnit/UnitTesting/Globalization/AvailableResourcesTester.cs
Comment (a-ctor) [reply SharedSource/Development/NUnit/UnitTesting/Globalization/AvailableResourcesTester.cs:19]: Linking does not work since a reference in the XML docs cannot be found. Including this file as well causes more XML doc issues and other compile errors because stuff could not be found.

**Review by MichaelKetting — CHANGES_REQUESTED** [review]
Existing stuff looks okay, now we need to put it into all the unit test for the projects that have resources.

**Comment by a-ctor** [comment]
@MichaelKetting we can do them now as it doesn't seem worth to me to open separate issues. There are 20 usages of the `AvailableResourcesLanguages`, i would need a list of what of those usages need tests and what do not.

**Comment by MichaelKetting** [comment]
> @MichaelKetting we can do them now as it doesn't seem worth to me to open separate issues. There are 20 usages of the `AvailableResourcesLanguages`, i would need a list of what of those usages need tests and what do not.

SecurityManagerClientsWeb (needs also a test project)
objectbindingwebcompatiblity 
objectbindingwebvalidation
objectbindingwebpreview (needs also a test project)

**Comment by a-ctor** [comment]
@MichaelKetting And what about the ones that have empty `AvailableResourceLanguages`? Which of those should I add the test to?

**Review by MichaelKetting — APPROVED** [review]

**Pull request merged**
